src/startupscriptgenerator/src/common/preRunScriptHelper.go

```diff
@@ -0,0 +1,22 @@
+// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+package common
+
+import (
+	"os"
+	"strings"
+	"common/consts"
+)
+
+// Run pre-run script from root of the app directory
+
+func SetupPreRunScript(scriptBuilder *strings.Builder)  {
+	preRunCommandOrScript := os.Getenv(consts.PreRunCommandEnvVarName)
+	if preRunCommandOrScript != "" {
+		scriptBuilder.WriteString("echo 'Running the provided pre-run command...'\n")
+		scriptBuilder.WriteString(preRunCommandOrScript + "\n")
```

Comment:
how do we know the path of the script? do we assume absolute path to be provided or do we support relative path as well?

Reply:
It could be error prone for us to parse the contents of this environment variable since user could provide a command, list of commands, mix commands and file paths etc. In my opinion we should add documentation suggesting users to have relative paths to scripts.

Reply:
agreed …. we should have doc related to that .. but at the basic level we could check as well whether the content of the env variable is a file or not a file. what do you think? @kichalla

Reply:
You mean if somebody just has `./prerun-script.sh` or `scripts/prerun-script.sh`, we could resolve them to full paths and run? Sure, we could try doing that but I do not see much value in it to be honest. If we can guarantee and document it that the user that the app's run context is always going to be root of the directory, then it should just work. Are there any other cases?

Reply:
I think, from customer point of view it doesn't add value much. But from tooling side if we know what user was trying to run (it's a command or it's a script file) it will be easier for us to troubleshoot seeing the log. End to end scenario wise I agree there is not much value, but adding a small check won't hurt us.

Reply:
> So if we check the file and error it out at-least user will know from basic context.

Sure I agree this would help the user.

Reply:
@arroyc for now I'm leaning to assume the path to be relative and provide a documentation for users, that'll save us some checks. And yes I'll add a check to see if the environment variable is a file or commands, good for users to see some context.

Reply:
An user case could be like `./prerun-script.sh && touch sample.txt`, we could not simply check if the environment variable is a path or not, the combination may also exist.

Reply:
should we let user to do that? we also have userstartup command to run commands. 
> Enabling to use a pre-run script to run command before the app is started.
Set the script path as an environment variable

As per the description and the test you had, I thought it will always be script path, if t is not then we have a bit different story here. But the use case you mentioned can you please add a relevant test.

Reply:
FYI: this environment variable can allow a command, list of commands, a script path, mix of these etc.